js/ai/src/document.ts

```diff
@@ -36,8 +51,76 @@ export const MediaPartSchema = EmptyPartSchema.extend({
     url: z.string(),
   }),
 });
+
+/**
+ * Media part.
+ */
 export type MediaPart = z.infer<typeof MediaPartSchema>;
 
+/**
+ * Zod schema of a tool request part.
+ */
+export const ToolRequestPartSchema = EmptyPartSchema.extend({
```

Comment:
wait.... why not import these from model.ts?

Reply:
Model imports DocumentData from document.js which causes a cyclic dependency.

Reply:
lets put them in a separate file them (parts.ts?)... document.ts is not a logic place for them.

Reply:
Hmm, I thought of that but `parts.ts` has no meaningful purpose to be its own file and decided to do this instead. I guess we can move things around since they are not exposed directly anyway

Reply:
Resolving, leaving as it is.